src/main/java/com/tlswe/awsmock/ec2/control/MockEC2QueryHandler.java

```diff
@@ -302,6 +435,11 @@ private DescribeInstancesResponseType describeInstances(final Set<String> instan
         for (AbstractMockEc2Instance instance : instances) {
 
             if (null != instance) {
+                String instanceState = instance.getInstanceState().getName();
+                // get instances with specified states
+                if (!instanceStates.isEmpty() && !instanceStates.contains(instanceState)) {
```

Comment:
Good code here! This maximizes the compatibility for all the old calls without states filter passed in:)